src/quo2/components/list_items/preview_list/view.cljs

```diff
@@ -0,0 +1,150 @@
+                                            :size              user-avatar-size}
+                                           item)]
+
+      :accounts                    [account-avatar/view
+                                    (merge item {:size size})]
+
+      (:communities :collectibles) [fast-image/fast-image
+                                    {:source (:source item)
+                                     :style  {:width         size
+                                              :height        size
+                                              :border-radius border-radius}}]
+
+      (:tokens :network :dapps)    [fast-image/fast-image
+                                    {:source (:source item)
+                                     :style  {:width         size
+                                              :height        size
+                                              :border-radius border-radius}}]
+      nil)))
+
+(defn- list-item
+  [{:keys [index type size-key item number]}]
+  (let [last-item?  (= index (- number 1))
+        margin-left (get-in properties/sizes [size-key :margin-left])
+        hole-size   (get-in properties/sizes [size-key :hole-size])
+        hole-radius (get-in properties/sizes [size-key :hole-radius (properties/border-type type)])
+        hole-x      (get-in properties/sizes [size-key :hole-x])
+        hole-y      (get-in properties/sizes [size-key :hole-y])]
+    [hole-view/hole-view
+     {:style {:margin-left (if (= index 0) 0 margin-left)}
+      :holes (if last-item?
+               []
+               [{:x            hole-x
+                 :y            hole-y
+                 :width        hole-size
+                 :height       hole-size
+                 :borderRadius hole-radius}])}
+     [preview-item
+      {:item     item
+       :type     type
+       :size-key size-key}]]))
+
+(defn- view-internal
+  "[preview-list opts items]
+   opts
+   {:type          :user/:communities/:accounts/:tokens/:collectibles/:dapps/:network
+    :size          :size/s-32 | :size/s-24 | :size/s-20 | :size/s-16 | :size/s-14
+    :number        number of items in the list (optional)
+    :blur?         overflow-label blur?}
+   items           preview list items (only 4 items is required for preview)
+  "
+  [{:keys [type size number blur? theme more-than-99-label]} items]
+  (let [size-key      (if (contains? properties/sizes size) size :size/s-24)
+        items-arr     (into [] items)
+        number        (or number (count items))
+        margin-left   (get-in properties/sizes [size-key :margin-left])
+        border-radius (get-in properties/sizes [size-key :border-radius (properties/border-type type)])]
+    [rn/view {:style {:flex-direction :row}}
+     (for [index (range (if (> number 4) 3 number))]
+       ^{:key (str index number)}
+       [list-item
+        {:index    index
+         :type     type
+         :size-key size-key
+         :item     (get items-arr index)
```

Comment:
We don't need that `into []` call, since items is already a sequence, just use `vec`, also `items-arr` is just being used once, so maybe it's enough if we remove that binding `items-arr` and just change:

`:item (get items-arr index`
for:
`:item (get (vec items) index)`